### PR TITLE
[SITE-2457] Reload core during post schema.

### PR DESCRIPTION
**Pull request by kporras07**
- Reloads schema after schema is posted
- Adds reload terminus command
- calls new reload endpoint as part of `reloadCore()` instead of the no-op log.

### CODE REVIEW

File: src/Services/Endpoint.php [hunk 1]
```diff
@@ -198,6 +198,22 @@ public function getSchemaUploadUri(): string {
...
   }
 
+  /**
+   * Get URL in pantheon environment to upload schema files.
+   *
+   * @return string
+   *   URL of envrionment.
+   */
+  public function getReloadUri(): string {
+    return vsprintf(
```
Comment (kporras07) [inline src/Services/Endpoint.php:208]: this function should be reimplemented

File: src/Services/SchemaPoster.php [hunk 2]
```diff
@@ -105,9 +106,55 @@ public function postSchema(string $server_id, $files = []): array {
...
     }
 
+    $status_code = $response->getStatusCode();
+    $this->logger->info('Status code: ' . $status_code);
+    if ($status_code >= 200 && $status_code < 300) {
+      // @TODO Maybe we need to capture exception here??
+      // Call reload on the server.
+      $this->reloadServer();
+    }
+    $this->logger->info('After server reload?');
```
Comment (kporras07) [inline src/Services/SchemaPoster.php:118]: extra debugging logging we should remove

File: src/Services/Endpoint.php (hunk 1 above)
Comment (stovak) [reply src/Services/Endpoint.php:208]: Kevin is correct. You'll need to get the reload URL from the env var.

File: src/Services/SchemaPoster.php (hunk 2 above)
Comment (stovak) [reply src/Services/SchemaPoster.php:118]: I would leave ALL of the debugging in until you have a release ready. Also I could do a drush command to call the reload as well.

**Review by stovak — DISMISSED** [review]
I think you need two things... 1. Change the way it gets the reload URL to the environment variable. 2. do a command to do the reload from drush.

**Comment by stovak** [comment]
Also having a drush command will help you debug the process once we get to testing. Once you know the drush command works and you can trigger it at will, you can know that as long as the UI is triggering that function, all is working as it should.

**Comment by kporras07** [comment]
I'm unsure whether reloading as a standalone command is needed/makes sense. AFAIK the goal is to make reload happen automatically on schema post

File: tests/Unit/EndpointServiceTest.php [hunk 3]
```diff
@@ -43,6 +44,19 @@ public function testURIGeneration() {
       'one://two:1234/server-path/schema-path',
       $ep->getSchemaUploadUri()
     );
+    $this->assertEquals(
+      'one://two:1234/server-path/schema-path',
+      $ep->getSchemaUploadUri()
+    );
```
Comment (pwtyler) [inline tests/Unit/EndpointServiceTest.php]:
```suggestion
```

File: src/Commands/Reload.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+   * @command search-api-pantheon:reloadSchema
+   */
+  public function reloadSchema() {
+    try {
+      $this->schemaPoster->reloadServer();
+    }
+    catch (\Exception $e) {
+      $this->logger->error((string) $e);
+    }
+  }
```
Comment (pwtyler) [inline src/Commands/Reload.php:66]: This currently doesn't display anything, not sure the idiomatic way to write a drush command so this is bare bones.

File: .gitignore [hunk 5]
```diff
@@ -8,3 +8,5 @@
 .idea
 .DS_Store
 docs/Badge.confluence
+
+.envrc
```
Comment (pwtyler) [inline .gitignore]:
```suggestion
.envrc

```

**Review by pwtyler — DISMISSED** [review]
Been a number of hands on this. +1 to cut a release and get a fix out with the caveat that unit tests are failing (and were failing before this PR) and units are not running in CI.

**Comment by pwtyler** [comment]
I've backed out the changes to PantheonSolrConnector (a6b1488 and its subsequent fixes), as we encountered an exception thrown from the dash running reload core at `/admin/config/search/search-api/server/pantheon_blog_solr8/solr-admin/reload-core`.

```
Message	RuntimeException: Unable to instantiate Schema Poster. in Drupal\search_api_pantheon\Plugin\SolrConnector\PantheonSolrConnector->reloadCore() (line 409 of /code/web/modules/contrib/search_api_pantheon/src/Plugin/SolrConnector/PantheonSolrConnector.php).
```

`drush search-api-pantheon:postSchema` and `/admin/config/search/search-api/server/pantheon_solr8/pantheon-admin/schema` seem fine though. still adovcating Reload Core may be best as follow-on work unless there's an important gap I'm missing here.

**Review by pwtyler — DISMISSED** [review]

**Review by namespacebrian — DISMISSED** [review]

**Comment by pwtyler** [comment]
Replacing this with #192. `git diff c4383b0 65a5c94` shows no difference. That PR will be our workspace for future iteration on the currently broken "reload core".

**Pull request closed without merging**